File: src/business/rule/engine/service/IUserRepository.java
package business.rule.engine.service;

import business.rule.engine.model.Agent;
import business.rule.engine.model.User;

public interface IUserRepository {
	 User GetUser(String userName);
     boolean AddUser(User user);
     boolean UpdateUser(User user);
     boolean UpdateAgent(Agent agent);
}

File: src/IUserRepository.cs
using Raymaker.RulesEngine.Application.Model;

namespace Raymaker.RulesEngine.Application
{
    public interface IUserRepository
    {
        User GetUser(string userName);
        bool AddUser(User user);
        bool UpdateUser(User user);
        bool UpdateAgent(Agent agent);
    }
}
